packages/telemetry/src/telemetryReporter.ts

```diff
@@ -192,7 +193,17 @@ export class AppInsights extends AsyncCreatable<TelemetryOptions> {
    */
   private buildContextTags(): Properties {
     const currentTags = this.appInsightsClient ? this.appInsightsClient.context.tags : {};
-    return Object.assign({}, currentTags, this.options.contextTags);
+    const cleanedTags = this.hideGDPRdata(currentTags);
+    return Object.assign({}, cleanedTags, this.options.contextTags);
+  }
+  // filters out non-GDPR compliant tags
+  private hideGDPRdata(tags: Properties) {
+    const keys = new appInsights.Contracts.ContextTagKeys();
+    const gdprSensitiveKeys = [keys.cloudRoleInstance];
+    gdprSensitiveKeys.forEach(key => {
+      tags[key] = AppInsights.GDPR_HIDDEN;
```

Comment:
Is this better than just removing it?

I think this should also be configurable... since Web IDE has started to use this for their API server, the instance is a completely valid thing to log. Not sure the right way to do that though. Maybe a property that says whether this runs on a customer's machine or not...

Reply:
To answer my own question, I think I like the `<GDPR_HIDDEN>` better than just removing it as well.